Microsoft.Toolkit.Uwp.UI.Controls/MosaicControl/MosaicControl.cs

```diff
@@ -0,0 +1,1098 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Numerics;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Robmikh.CompositionSurfaceFactory;
+    using Windows.ApplicationModel;
+    using Windows.Foundation;
+    using Windows.Foundation.Metadata;
+    using Windows.Storage;
+    using Windows.UI.Composition;
+    using Windows.UI.Xaml;
+    using Windows.UI.Xaml.Controls;
+    using Windows.UI.Xaml.Hosting;
+    using Windows.UI.Xaml.Media;
+    using Windows.UI.Xaml.Media.Imaging;
+    using Windows.UI.Xaml.Shapes;
+
+    /// <summary>
+    /// Orientation of the scroll
+    /// </summary>
+    public enum ScrollOrientation
+    {
+        /// <summary>
+        /// Scroll only Horizontaly (and optimize the number of image used)
+        /// </summary>
+        Horizontal,
+
+        /// <summary>
+        /// Scroll only Verticaly (and optimize the number of image used)
+        /// </summary>
+        Vertical,
+
+        /// <summary>
+        /// Scroll both Horizontaly and verticaly
+        /// </summary>
+        Both
+    }
+
+    public enum ImageAlignment
+    {
+        /// <summary>
+        /// No alignment needed
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// Align to Left when the property ScrollOrientation is Horizontal
+        /// </summary>
+        Left,
+
+        /// <summary>
+        /// Align to Right when the property ScrollOrientation is Horizontal
+        /// </summary>
+        Right,
+
+        /// <summary>
+        /// Align to Top when the property ScrollOrientation is Vertical
+        /// </summary>
+        Top,
+
+        /// <summary>
+        /// Align to Bottom when the property ScrollOrientation is Vertical
+        /// </summary>
+        Bottom
+    }
+
+    /// <summary>
+    /// A ContentControl that show an image repeated many times.
+    /// The control can be synchronized with a Scrollviewer and animated easily.
+    /// </summary>
+    public sealed class MosaicControl : ContentControl
+    {
+        // Using a DependencyProperty as the backing store for ScrollViewer.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ScrollViewerContainerProperty =
+            DependencyProperty.Register("ScrollViewerContainer", typeof(FrameworkElement), typeof(MosaicControl), new PropertyMetadata(null, OnScrollViewerContainerChange));
```

Comment:
Inline with the style of some of the other controls in the project, you may want to consider `DependencyProperty.Register(nameof(ScrollViewerContainer)....` rather than `DependencyProperty.Register("ScrollViewerContainer"...`

(same applies for the rest of the DP's)